requirements/docs.txt

```diff
@@ -6,6 +6,7 @@ sphinx-gallery
 sphinx-copybutton
 pytest-runner
 scikit-learn
+matplotlib>=3.1.0
```

Comment:
can you explain the need to go bleeding edge here?

Reply:
Sure, the issue is that `constrained_layout` isn't implemented in matplotlib 2.0 so one of the doc builds was failing. `constrained_layout` is implemented in 2.2 but is experimental. It received a substantial update in matplotlib 3.0 to also constrain text:
https://matplotlib.org/3.0.0/api/api_changes.html
```
Layout tools like Figure.tight_layout, constrained_layout, and fig.savefig('fname.png', bbox_inches="tight") use matplotlib.axes.Axes.get_tightbbox to determine the bounds of each axes on a figure and adjust spacing between axes.

In Matplotlib 2.2 get_tightbbox started to include legends made on the axes, but still excluded some other artists, like text that may overspill an axes. This has been expanded to include all artists.
```
@jni suggested 3.1.0 so I went with that over 3.0.

Reply:
I guess it is the docs so it is fine. I'm cool with bleeding edge anyway. Much more fun :D.